mapie/conformity_scores.py

```diff
@@ -136,9 +143,23 @@ def check_consistency(
                 "sure that the two methods are consistent."
             )
 
-    def get_conformity_scores(
-        self, y: ArrayLike, y_pred: ArrayLike
-    ) -> NDArray:
+    def get_weights(self, X: NDArray) -> NDArray:
+        """
+        Compute weights for conformity scores on calibration samples.
+
+        Parameters
+        ----------
+        X : NDArray
+            Dataset used for computing the weights
+
+        Returns
+        -------
+        NDArray
+            Estimated weights
+        """
+        return np.ones(shape=(len(X) + 1))
```

Comment:
As I understand it, the `get_weights` method is only used in the `CovariateShiftConformityScore` class. It is not used in the already implemented `ConformityScore` classes. From this observation:

-  I suggest you implement a new `WeightedConformalScore` class (in a new file `weighted_conformity_scores.py`), which inherits from `ConformalScore` with all your current changes.
- Thus the `CovariateShiftConformityScore` class will inherit from `WeightedConformalScore` and will be the only class that benefits from your changes without impacting the others.

I can confirm that this new structure will have an impact on the future features we plan to implement. Indeed, we plan to implement conditional conformal predictions, and weighted conformal scores will be the first step in this direction, thanks to you.